cli_047_ndc_ratification/contents/src/__init__.py

```diff
@@ -7,99 +7,258 @@
+        for f in os.listdir(DATA_DIR):
+            # try to remove it as a file
+            try:
+                logging.info('Removing {}'.format(f))
+                os.remove(DATA_DIR+'/'+f)
+            # if it is not a file, remove it as a folder
+            except:
+                shutil.rmtree(f)
+    except NameError:
+        logging.info('No local files to clean.')
+
+'''
+FUNCTIONS FOR THIS DATASET
+
+The functions below have been tailored to this specific dataset.
+They should all be checked because their format likely will need to be changed.
+'''
+
+def processNewData(SOURCE_URL):
+    '''
+    Fetch, process and upload new data
+    INPUT   SOURCE_URL: url where you can find the download link for the source data (string)
+    RETURN  num_new: number of rows of new data sent to Carto table (integer)
+    '''
+    # specify the starting page of source url we want to pull
     page = 1
+    # generate the url to pull data for this page 
     r = requests.get(SOURCE_URL.format(page=page))
+    # pull data from request response json
     raw_data = r.json()['data']
+    # if data is available from source url 
     if len(raw_data)>0:
-        if cartosql.tableExists(CARTO_TABLE):
-            cartosql.deleteRows(CARTO_TABLE, 'cartodb_id IS NOT NULL', user=os.getenv('CARTO_USER'), key=os.getenv('CARTO_KEY'))
+        # if the table exists
+        if cartosql.tableExists(CARTO_TABLE, user=CARTO_USER, key=CARTO_KEY):
+            # delete all the rows
+            cartosql.deleteRows(CARTO_TABLE, 'cartodb_id IS NOT NULL', user=CARTO_USER, key=CARTO_KEY)
         logging.info('Updating {}'.format(CARTO_TABLE))
     else:
+        # raise an error that data is not available from source url 
         logging.error("Source data missing. Table will not update.")
-    new_rows = []
+    # create an empty list to store new data
+    new_data = []
+    # if data is available from source url 
     while len(raw_data)>0:
         logging.info('Processing page {}'.format(page))
+        # read in source data as a pandas dataframe
         df = pd.DataFrame(raw_data)
+        # go through each rows in the dataframe
         for row_num in range(df.shape[0]):
+            # get the value from current row
             row = df.iloc[row_num]
+            # create an empty list to store data from this row
             new_row = []
+            # go through each column in the Carto table
             for field in CARTO_SCHEMA:
+                # if we are fetching data for unique id column
                 if field == 'uid':
+                    # add the unique id to the list of data from this row
                     new_row.append(row[UID_FIELD])
+                # for any other column, check if there are values available from the source for this row    
                 else:
+                    # if data available from source for this field, populate the field with the data
+                    # else populate with 'None' 
```

Comment:
just changing this because it sort of looks like None is a string when you put it in quotes, but it is not
```suggestion
                    # else populate with None
```